### ISSUE TITLE
how to covert deep images to surface normal images 

### ISSUE THREAD

**Issue by ITking666** [ialhashim/DenseDepth#131]
Dear ialhasim:
     would please tell me that how to covert deep images to surface normal images .Can you share your codes for converting.

**Comment by ialhashim** [comment]
Try this code snippet:
```
import numpy as np

def compute_normals(y):
    dx, dy = np.gradient(y)
    dz = np.ones_like(dy)
    dx = dx * 25
    dy = dy * 25
    n = np.linalg.norm(np.stack((dy, dx, dz), axis=-1), axis=-1)
    d = np.stack((dy/n, dx/n, dz/n), axis=-1)
    return (d + 1) * 0.5
```

**Comment by ITking666** [comment]
Dear&nbsp;Sir:
Thank you for your kind reply


Best wishes





------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"notifications"<notifications@github.com&gt;;
发送时间:&nbsp;2020年6月14日(星期天) 凌晨4:43
收件人:&nbsp;"ialhashim/DenseDepth"<DenseDepth@noreply.github.com&gt;;
抄送:&nbsp;"1474690467"<1474690467@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [ialhashim/DenseDepth] how to covert deep images to surface normal images  (#131)





 
Closed #131.
